tests/integration/targets/virtual_disk_attach/tasks/01_supported.yml

```diff
@@ -0,0 +1,246 @@
+      - block_device is changed
+      - >
+        block_device.record.keys() | sort == ['cache_mode', 'disable_snapshotting', 'disk_slot', 'iso_name',
+        'mount_points', 'read_only', 'size', 'tiering_priority_factor', 'type', 'uuid', 'vm_uuid']
+      - block_device.record.type == "virtio_disk"
+      - block_device.record.disk_slot == 3
+      - block_device.record.size == 104857600
+      - block_device.record.iso_name == ""
+      - block_device.record.tiering_priority_factor == 8 # THIS DOESN'T ALWAYS WORK - RETURNS 0
+      - block_device.record.cache_mode == "writethrough"
+      - block_device.record.disable_snapshotting == true
+      - block_device.record.read_only == true
+      - block_device.record.vm_uuid == vm.record.0.uuid 
+      - not block_device.diff.before
+      - block_device.diff.after == block_device.record
+  
+- name: Assert that disk is attached to XLAB-attach-virtual-disk
+  scale_computing.hypercore.vm_info:
+    vm_name: XLAB-attach-virtual-disk
+  register: vm
+- ansible.builtin.assert:
+    that:
+      - vm.records.0.vm_name == "XLAB-attach-virtual-disk"
+      - vm.records.0.disks | length == 1
+      - vm.records.0.disks.0.type == "virtio_disk"
+      - vm.records.0.disks.0.disk_slot == 3
+      - vm.records.0.disks.0.size == 104857600
+      - vm.records.0.disks.0.size > virtual_disk.records.0.size 
+      - vm.records.0.disks.0.iso_name == ""
+      - vm.records.0.disks.0.tiering_priority_factor == 8 # THIS DOESN'T ALWAYS WORK - RETURNS 0
+      - vm.records.0.disks.0.cache_mode == "writethrough"
+      - vm.records.0.disks.0.disable_snapshotting == true
+      - vm.records.0.disks.0.read_only == true
+      - vm.records.0.disks.0.uuid == block_device.record.uuid
+      - vm.records.0.disks.0.vm_uuid == block_device.record.vm_uuid
+
+- name: Attach virtual disk in slot 3 as a virtio_disk - IDEMPOTENCE
+  scale_computing.hypercore.virtual_disk_attach: *attach
+  register: block_device
+- ansible.builtin.assert: 
+    that:
+      - block_device is succeeded
+      - block_device is not changed
+      - >
+        block_device.record.keys() | sort == ['cache_mode', 'disable_snapshotting', 'disk_slot', 'iso_name',
+        'mount_points', 'read_only', 'size', 'tiering_priority_factor', 'type', 'uuid', 'vm_uuid']
+      - block_device.record.type == "virtio_disk"
+      - block_device.record.disk_slot == 3
+      - block_device.record.size == 104857600
+      - block_device.record.iso_name == ""
+      - block_device.record.tiering_priority_factor == 8 # THIS DOESN'T ALWAYS WORK - RETURNS 0
+      - block_device.record.cache_mode == "writethrough"
+      - block_device.record.disable_snapshotting == true
+      - block_device.record.read_only == true
+      - block_device.record.vm_uuid == vm.records.0.uuid
+      - block_device.diff.after == block_device.record
+      - block_device.diff.before == block_device.record
+
+- name: Attach virtual disk with only type defined (test default values)
```

Comment:
True, it will create disk in the next empty slot if the slot is not defined. Slot is not required in payload, thus I didn't require it in the module, but it makes sense to be required. I updated the module.